lib/rust/ensogl/core/src/debug/monitor.rs

```diff
@@ -191,34 +295,170 @@ impl Drop for DomData {
 }
 
 
+// === MainArea ===
+
+/// The main area (left panel) of the performance monitor. It contains samplers names, samplers
+/// values, and value plots. It also contains play/pause button and the visual selection region.
+#[derive(Debug, Deref)]
+pub struct MainArea {
+    #[deref]
+    root:         web::HtmlDivElement,
+    plot:         web::HtmlCanvasElement,
+    plot_context: web::CanvasRenderingContext2d,
+    selection:    web::HtmlDivElement,
+}
+
+impl MainArea {
+    fn new(config: &Config) -> Self {
+        let root = web::document.create_div_or_panic();
+        root.set_style_or_warn("overflow", "hidden");
+        root.set_style_or_warn("border-radius", "6px");
+        root.set_style_or_warn("border", "2px solid #000000c4");
+        root.set_style_or_warn("box-sizing", "content-box");
+        root.set_style_or_warn("position", "relative");
+
+        let plot = web::document.create_canvas_or_panic();
+        plot.set_style_or_warn("display", "block");
+        root.append_child(&plot).unwrap();
+
+        let plot_context = plot.get_context("2d").unwrap().unwrap();
+        let plot_context: web::CanvasRenderingContext2d = plot_context.dyn_into().unwrap();
+
+        let selection_border = config.plot_selection_border;
+        let selection_width = config.plot_selection_width + 2.0 * selection_border;
+        let selection_left =
+            config.plot_right_x() - 0.5 - config.plot_selection_width / 2.0 - selection_border;
```

Comment:
What's `0.5` here?

Reply:
It's centering of selection. I refactored the code to be more obvious and moved it to 
```
let selection_center = (config.plot_selection_width + 1.0) / 2.0;
```

Is it ok now?